### PR TITLE
Upgrade concat

### PR DESCRIPTION
**Pull request by sellout**
Removes our dependency on our fork of concat.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -47,7 +51,7 @@ constraints:
   , either >=5.0.1 && <5.1
   , extra >=1.7.8 && <1.8
   , fin >=0.1.1 && <0.3
-  , ghc >=0.10.1 && <8.11
+  , ghc >=8.10.2 && <8.10.7
```
Comment (zliu41) [inline cabal.project:54]: The lower bound should be 8.10.1.

Comment (sellout) [reply cabal.project]: Fixed

File: integrations/kitty/category/Kitty/Plugin/Kitty.hs [hunk 2]
```diff
@@ -185,50 +173,24 @@ class SemigroupCat (k :: Type -> Type -> Type) m where
 instance Semigroup m => SemigroupCat (->) m where
   appendK = uncurry (<>)
 
--- | A traced Cartesian category provides a notion of a parameterized fixed-point operator. There is
```
Comment (zliu41) [inline integrations/kitty/category/Kitty/Plugin/Kitty.hs:188]: These comments seem helpful. Are they no longer applicable? Also why the renaming?

Comment (sellout) [reply integrations/kitty/category/Kitty/Plugin/Kitty.hs:188]: It's not a renaming. We used to have both TracedCat and FixedCat, but turns out there's actually TracedCat already in concat-classes, so I just removed our TracedCat, but left our FixedCat in place.

I agree there's useful stuff in the comment (that comment is mostly copied from Haskerwaul, actually), but the part about "Currently, we can't generate @trace@ during conversion, so we don't bother to separate the classes." is false -- we generate it from ArrowLoop, and we do separate the classes (this comment was on FixedCat before we split out TracedCat).

We just don't have a TracedCat to attach the remainder of the comment to.

**Pull request merged**